nautobot_firewall_models/capirca_utils.py

```diff
@@ -0,0 +1,415 @@
+                    key = key[4:]
+                if not value:
+                    continue
+                if isinstance(value, list):
+                    for item in value:
+                        if not item:
+                            continue
+                        pol.append(f"  {key}:: {item}")
+                        LOGGER.debug("Adding term key: `%s` value: `%s`", str(key), str(item))
+                else:
+                    pol.append(f"  {key}:: {value}")
+                    LOGGER.debug("Adding term key: `%s` value: `%s`", str(key), str(value))
+            pol.append("}")
+            pol.append("")
+
+        networkdata = []
+        for name, addresses in self.address.items():
+            addresses = _clean_list(addresses, True)
+            first_address = addresses.pop(0)
+            networkdata.append(f"{name} = {first_address}")
+            for address in addresses:
+                networkdata.append(" " * (len(name) + 3) + address)
+
+        servicedata = []
+        for name, services in self.service.items():
+            services = _clean_list(services, True)
+            first_service = services.pop(0)
+            servicedata.append(f"{name} = {first_service}")
+            for service in services:
+                servicedata.append(" " * (len(name) + 3) + service)
+
+        self.pol_file = "\n".join(pol)
+        self.svc_file = "\n".join(servicedata)
+        self.net_file = "\n".join(networkdata)
+
+        self.cfg_file = generate_capirca_config(servicedata, networkdata, self.pol_file, self.platform)
+
+
+class DevicePolicyToCapirca(PolicyToCapirca):
+    """Class object to convert Policy orm object to Capirca object for a whole device."""
+
+    def __init__(self, device_obj, *args, **kwargs):  # pylint: disable=super-init-not-called
+        """Overload init to account for computed field."""
+        self.policy_name = ""
+        self.platform_original = device_obj.platform.slug
+        self.policy_objs = []
+        self.policy_obj = ""
+        self.platform = CAPIRCA_OS_MAPPER.get(device_obj.platform.slug, device_obj.platform.slug)
+        self.address = {}
+        self.service = {}
+        self.policy = []
+        policy_name = []
+        self.pol_file = ""
+        self.svc_file = ""
+        self.net_file = ""
+        self.cfg_file = ""
+        _allow_list = device_obj.platform.custom_field_data.get("capirca_allow")
+        self.cf_allow_list_enabled = True if "capirca_allow" in device_obj.platform.custom_field_data else False
+        self.cf_allow_list = _allow_list if _allow_list else []
+        LOGGER.debug("Capirca Platform Name: `%s`", str(self.platform))
+        LOGGER.debug("Original Platform Name: `%s`", str(device_obj.platform.slug))
+        LOGGER.debug("cf_allow_list_enabled: `%s`", str(self.cf_allow_list_enabled))
+        LOGGER.debug("cf_allow_list: `%s`", str(self.cf_allow_list))
+
+        # TODO: Determine if this can be done with weight and more efficient
```

Comment:
@whitej6 to create a convenience method to provide this ordered via the weight.

Reply:
To be handled by #70